Feature/us12/graphique

JarodEphec opened the pull request:
# Description
La description de cette US est de disponible [ici](https://github.com/users/Martin-Hayot/projects/2?pane=issue&itemId=39939662)

Le But de cette US est de créé un composant réutilisable react qui permet d'afficher des graphiques avec les données que l'on lui envoie.

# Utilisation
Le composant s'appelle `LineChart` et est appelé de cette manière : 
```
<LineChart title={chartTitle} xTitle={xTitle} yTitle={yTitle} data={chartData} />
```
Les variables que prend le composant sont : 

- title : Pour donner un titre au graphique
- xTitle :  Pour nommer l'axe x
- yTitle :  Pour nommer l'axe y
- data : Pour alimenter le graphique des données qui vont être affichés

La page `Chart` sert à montrer un exemple de l'utilisation des graphiques, cette page pourra être implémentée dans le site, j'attends d'en discuter avec vous :)

Samaze123 reviewed (changes requested):
## Page Chart

Belle page pour montrer comment un graphique fonctionne.

Petits problèmes :

- Pas de navbar sur la page

## Composant Graphique

Le composant est bien fait mais voici des changment a faire :

- Quand je redimensionne la fenetre, le graphique dépasse et est presque coupé de moitié.
- Ne pas être obligé d'indiquer des titres pour le composants (les rendres optionels).

## Divers

Tu m'as dit en appel que le fichier *aquarium.tsx* doit être supprimé, ne l'oublie pas.

***Sinon dans l'ensemble c'est une belle fonctionnalité !***

JarodEphec commented:
> ## Page Chart
> 
> Belle page pour montrer comment un graphique fonctionne.
> 
> Petits problèmes :
> 
>     * Pas de navbar sur la page
> 
> 
> ## Composant Graphique
> 
> Le composant est bien fait mais voici des changment a faire :
> 
>     * Quand je redimensionne la fenetre, le graphique dépasse et est presque coupé de moitié.
> 
>     * Ne pas être obligé d'indiquer des titres pour le composants (les rendres optionels).
> 
> 
> ## Divers
> 
> Tu m'as dit en appel que le fichier _aquarium.tsx_ doit être supprimé, ne l'oublie pas.
> 
> _**Sinon dans l'ensemble c'est une belle fonctionnalité !**_

Salut Samuel, merci pour la review, j'ai fait tous les changements que tu m'as demandés.

Pour ce qui est des arguments de titres, j'ai fait le test et si tu ne mets rien, rien n'est affiché là où il y aurait les titres habituellement, tout fonctionne normalement.

Samaze123 commented:
Je viens de voir et ***package-lock.json*** est commit, je ne sais pas si c'est volontaire ou pas.

J'ai essayé tes changements et tout me semble correct pour moi ! 

Très chouettes graphiques.

JarodEphec committed:
US 12 feat : Ajout du composant graphique et de la page pour voir les les données de la sonde sous forme de graphique.

Pour consulter les données brutes envoyées par la sonde, des graphiques seront utilisés pour rendre la compréhension des résultats de la sonde de manière plus visuelle.

Ces graphiques ne seront pas seulement utilisés dans la page de cette US, mais aussi pour être utilisé sous forme de React component.

Cette US a comme double role de montrer l'utilisation des graphiques dans d'autre partie du projet.

# Utilisation
Le composant s'appelle `LineChart` et est appelé de cette manière :
```
<LineChart title={chartTitle} xTitle={xTitle} yTitle={yTitle} data={chartData} />
```
Les variables que prend le composant sont :

- title : Pour donner un titre au graphique
- xTitle :  Pour nommer l'axe x
- yTitle :  Pour nommer l'axe y
- data : Pour alimenter le graphique des données qui vont être affichés
